config/src/inline/conf_parser.rs

```diff
@@ -0,0 +1,129 @@
+use std::{num::ParseIntError, str::ParseBoolError};
+
+use regex::Regex;
+
+/// Errors returned by the [`ConfParser`] trait.
+#[derive(Debug, Clone, PartialEq, Eq, thiserror::Error)]
+pub enum ConfParserError {
+    /// An invalid configuration property has been provided.
+    /// The property cannot be mapped to the configuration object
+    #[error("'{0}' is not a valid config property")]
+    InvalidConfigProperty(String),
+    /// An error occurred while trying to parse an integer configuration value
+    #[error(transparent)]
+    ParseIntError(#[from] ParseIntError),
+    /// An error occurred while trying to parse a boolean configuration value
+    #[error(transparent)]
+    ParseBoolError(#[from] ParseBoolError),
+}
+
+/// This trait is intended to parse configurations from
+/// structured text. Foundry users can annotate Solidity test functions,
+/// providing special configs just for the execution of a specific test.
+///
+/// An example:
+///
+/// ```solidity
+/// contract MyTest is Test {
+/// /// forge-config: default.fuzz.runs = 100
+/// /// forge-config: ci.fuzz.runs = 500
+/// function test_SimpleFuzzTest(uint256 x) public {...}
+///
+/// /// forge-config: default.fuzz.runs = 500
+/// /// forge-config: ci.fuzz.runs = 10000
+/// function test_ImportantFuzzTest(uint256 x) public {...}
+/// }
+/// ```
+pub trait ConfParser {
```

Comment:
nit: would like to rename this to make it clear that this is for inline config

```suggestion
pub trait InlineConfigParser {
```